### PR TITLE
New SignedData with content type

### PR DESCRIPTION
**Pull request by bilalashraf123**

### CODE REVIEW

File: sign_test.go [hunk 1]
```diff
@@ -212,6 +212,32 @@ func ExampleSignedData() {
 	pem.Encode(os.Stdout, &pem.Block{Type: "PKCS7", Bytes: detachedSignature})
 }
 
+func TestSignedDataWithContentType(t *testing.T) {
+	// generate a signing cert or load a key pair
+	cert, err := createTestCertificate(x509.SHA256WithRSA)
+	if err != nil {
+		fmt.Printf("Cannot create test certificates: %s", err)
```
Comment (vanbroup) [inline sign_test.go]:
```suggestion
		t.Fatalf("Cannot create test certificates: %s", err)
```

File: sign_test.go [hunk 2]
```diff
@@ -212,6 +212,32 @@ func ExampleSignedData() {
...
+	}
+
+	// Initialize a SignedData struct with content to be signed
+	signedData, err := NewSignedDataWithContentType(asn1.ObjectIdentifier{1, 2, 840, 113549, 1, 9, 16, 1, 4}, []byte("Example data to be signed"))
+	if err != nil {
+		fmt.Printf("Cannot initialize signed data: %s", err)
```
Comment (vanbroup) [inline sign_test.go]:
```suggestion
		t.Fatalf("Cannot initialize signed data: %s", err)
```

File: sign_test.go [hunk 3]
```diff
@@ -212,6 +212,32 @@ func ExampleSignedData() {
...
+	}
+
+	// Add the signing cert and private key
+	if err := signedData.AddSigner(cert.Certificate, cert.PrivateKey, SignerInfoConfig{}); err != nil {
+		fmt.Printf("Cannot add signer: %s", err)
```
Comment (vanbroup) [inline sign_test.go]:
```suggestion
		t.Fatalf("Cannot add signer: %s", err)
```

File: sign_test.go [hunk 4]
```diff
@@ -212,6 +212,32 @@ func ExampleSignedData() {
...
+	}
+
+	// Finish() to obtain the signature bytes
+	detachedSignature, err := signedData.Finish()
+	if err != nil {
+		fmt.Printf("Cannot finish signing data: %s", err)
```
Comment (vanbroup) [inline sign_test.go]:
```suggestion
		t.Fatalf("Cannot finish signing data: %s", err)
```

File: sign_test.go (hunk 1 above)
Comment (bilalashraf123) [reply sign_test.go]: Done.

File: sign_test.go (hunk 2 above)
Comment (bilalashraf123) [reply sign_test.go]: Done.

File: sign_test.go (hunk 3 above)
Comment (bilalashraf123) [reply sign_test.go]: Done.

File: sign_test.go (hunk 4 above)
Comment (bilalashraf123) [reply sign_test.go]: Done.

**Pull request merged**